code/obj/cryotron.dm

```diff
@@ -361,9 +363,58 @@
 		if (!exit_prompt(user))
 			return ..()
 
+	/// Handling dragging players in to cryo, mainly for silicon players.
 	MouseDrop_T(atom/target, mob/user as mob)
-		if (ishuman(target) && isrobot(user) && BOUNDS_DIST(src, user) == 0 && BOUNDS_DIST(src, target) == 0 && BOUNDS_DIST(user, target) == 0)
-			insert_prompt(target, user)
+		if (!ishuman(target) && !isrobot(target))
+			return
+
+		if (BOUNDS_DIST(src, user) != 0)
+			return
+
+		if (BOUNDS_DIST(src, target) != 0)
 			return
+
+		if (BOUNDS_DIST(user, target) != 0)
+			return
+
+		insert_prompt(target, user)
 		return ..()
 
+	/// Override for handling all interactions with the cryo chamber
+	Click()
+		if (!usr)
+			return
+
+		if (isAIeye(usr))
+			boutput(usr, "<span class='alert'>An incorporeal manifestation of an artificial intelligence's presence can't enter \the [src]!</span>")
+			return
+
+		if (!can_act(usr))
+			return
+
+		if (!in_interact_range(src, usr))
+			return
+
+		if (isdead(usr))
+			return
+
+		if (isobserver(usr))
+			return
+
+		if (isintangible(usr))
+			return
+
+		if (issilicon(usr))
+			enter_prompt(usr)
+
+		var/obj/item/in_hand_item = usr.equipped()
+
+		if (in_hand_item != null)
+			src.cryo_attackby(in_hand_item, usr)
+
+		else if (usr.equipped_limb() != null && in_hand_item == null)
+			src.cryo_attack_hand(usr)
+		else
+			enter_prompt(usr)
+
+		. = ..()
```

Comment:
kind of strange to put this at the end? `. = ..()` means roughly "return `..()` when the proc ends", so you could just put `..()` instead.